src/file.ts

```diff
@@ -1802,25 +1804,32 @@ class File extends ServiceObject<File> {
       const configDir = xdgBasedir.config || os.tmpdir();
 
       fs.access(configDir, fs.constants.W_OK, err => {
-        if (err) {
-          if (options.resumable) {
-            const error = new ResumableUploadError(
-              [
-                'A resumable upload could not be performed. The directory,',
-                `${configDir}, is not writable. You may try another upload,`,
-                'this time setting `options.resumable` to `false`.',
-              ].join(' ')
-            );
-            stream.destroy(error);
-            return;
+        const maybeCreateFolder = async () => {
+          if (err) {
+            try {
+              await mkDirAsync(configDir, {mode: 0o0700});
+            } catch (mkDirErr) {
+              if (options.resumable) {
+                const error = new ResumableUploadError(
+                  [
+                    'A resumable upload could not be performed. The directory,',
+                    `${configDir}, is not writable. You may try another upload,`,
+                    'this time setting `options.resumable` to `false`.',
+                  ].join(' ')
+                );
+                stream.destroy(error);
+                return;
+              }
+              // User didn't care, resumable or not. Fall back to simple upload.
+              this.startSimpleUpload_(fileWriteStream, options);
+              return;
+            }
           }
-
-          // User didn't care, resumable or not. Fall back to simple upload.
-          this.startSimpleUpload_(fileWriteStream, options);
-          return;
-        }
-
-        this.startResumableUpload_(fileWriteStream, options);
+          this.startResumableUpload_(fileWriteStream, options);
+        };
+        maybeCreateFolder().catch(e => {
+          throw new Error(e);
```

Comment:
This line is not covered by code coverage, but I'm not sure how to get to this case. This error shouldn't get thrown. Any ideas for how to write a test for this are welcome.

Reply:
Maybe related, do we need the function “maybeCreateFolder”? Can the code inside simply execute when the fs.access callback runs?

Reply:
Good point. Fixed!